genai_stack/stack/mediator.py

```diff
@@ -17,6 +17,22 @@ def get_embedded_text(self, text):
 
     def get_model_response(self, prompt: str) -> str:
         return self._stack.model.predict(prompt)
+    
+    # Memory component
+    def add_text(self, user_text:str, model_text:str) -> None:
+        self._stack.memory.add_text(user_text, model_text)
```

Comment:
You dont have to add all the methods of the memory component only methods that are needed in other components. And also do checks like if the component exists or not